core-java-modules/core-java-collections-6/src/test/java/com/baeldung/passcollectiontovarargs/PassCollectionToVarargsUnitTest.java

```diff
@@ -0,0 +1,40 @@
+package com.baeldung.passcollectiontovarargs;
+
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PassCollectionToVarargsUnitTest {
+
+    private List<String> listOfStrings;
+
+    @Test
+    void givenList_whenUsingCollectionToArray_thenInvokeVarargsMethod() {
+        method(listOfStrings.toArray(new String[0]));
+    }
+
+    @Test
+    void givenList_whenUsingForLoopToPopulateArray_thenInvokeVarargsMethod() {
+        String[] array = new String[listOfStrings.size()];
+        for (int i = 0; i < listOfStrings.size(); i++) {
+            array[i] = listOfStrings.get(i);
+        }
+        method(array);
+    }
+
+    @Test
+    void givenList_whenUsingStreamAPI_thenInvokeVarargsMethod() {
+        String[] array = listOfStrings.stream()
+            .toArray(String[]::new);
+        method(array);
+    }
```

Comment:
We need to have asserts for the tests.

Reply:
assert does not throw added to each junit test